From b18e795e2d50958938d5dfc81d3c1fc9d74cfad2 Mon Sep 17 00:00:00 2001
From: Simone Schwizer <simone.schwizer@bytesatwork.ch>
Date: Tue, 30 Sep 2025 10:08:24 +0200
Subject: [PATCH] configs: am62xx_bytedevkit: Add password

Enables autoboot stop password and sets boot delay to 1 second.

Signed-off-by: Simone Schwizer <simone.schwizer@bytesatwork.ch>
---
 configs/am62x_bytedevkit_a53_defconfig | 5 +++++
 configs/am62x_bytedevkit_r5_defconfig  | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/configs/am62x_bytedevkit_a53_defconfig b/configs/am62x_bytedevkit_a53_defconfig
index e10436e5df7..3d63084745e 100644
--- a/configs/am62x_bytedevkit_a53_defconfig
+++ b/configs/am62x_bytedevkit_a53_defconfig
@@ -35,6 +35,11 @@ CONFIG_ENV_VARS_UBOOT_CONFIG=y
 CONFIG_SPL_LOAD_FIT=y
 CONFIG_SPL_LOAD_FIT_ADDRESS=0x81000000
 CONFIG_SUPPORT_RAW_INITRD=y
+CONFIG_BOOTDELAY=1
+CONFIG_AUTOBOOT_KEYED=y
+CONFIG_AUTOBOOT_ENCRYPTION=y
+CONFIG_AUTOBOOT_STOP_STR_ENABLE=y
+CONFIG_AUTOBOOT_STOP_STR_SHA256="746ff992cd97391b15891f93dd1ce02908c33947c60f1a95fc134d40874e5ac0"
 CONFIG_USE_BOOTCOMMAND=y
 CONFIG_BOOTCOMMAND="run bsp_bootcmd"
 CONFIG_BOARD_LATE_INIT=y
diff --git a/configs/am62x_bytedevkit_r5_defconfig b/configs/am62x_bytedevkit_r5_defconfig
index 83ba657303a..f81aff5525b 100644
--- a/configs/am62x_bytedevkit_r5_defconfig
+++ b/configs/am62x_bytedevkit_r5_defconfig
@@ -30,6 +30,11 @@ CONFIG_SPL_LIBDISK_SUPPORT=y
 CONFIG_SPL_LOAD_FIT=y
 CONFIG_SPL_LOAD_FIT_ADDRESS=0x80080000
 CONFIG_LEGACY_IMAGE_FORMAT=y
+CONFIG_BOOTDELAY=1
+CONFIG_AUTOBOOT_KEYED=y
+CONFIG_AUTOBOOT_ENCRYPTION=y
+CONFIG_AUTOBOOT_STOP_STR_ENABLE=y
+CONFIG_AUTOBOOT_STOP_STR_SHA256="746ff992cd97391b15891f93dd1ce02908c33947c60f1a95fc134d40874e5ac0"
 # CONFIG_DISPLAY_CPUINFO is not set
 CONFIG_SPL_SIZE_LIMIT_SUBTRACT_GD=y
 CONFIG_SPL_SIZE_LIMIT_SUBTRACT_MALLOC=y
